### PR TITLE
Adds API v3.1 additions, changes base URLs, adds tests, bumps version

### PR DESCRIPTION
**Pull request by superdevx64**
* adds 'Download check' endpoint + test
* adds support for `webhook_ids`
* adds `privacy_notices_read_consent_given` to check object (no longer write only)
* bumps base URL versions to 3.1
* bumps version of library to 2.0.0

### CODE REVIEW

**Review by JakeBat — APPROVED** [review]
I added an import for FileDownload to the CheckManager class, but everything else looks good, approved.

File: onfido-java/src/test/java/com/onfido/webhooks/WebhookEventVerifierTest.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
   WebhookEventVerifier verifier = new WebhookEventVerifier(webhookToken);
 
   String rawEvent =
-      "{\"payload\":{\"resource_type\":\"check\",\"action\":\"check.completed\",\"object\":{\"id\":\"check-123\",\"status\":\"complete\",\"completed_at_iso8601\":\"2020-01-01T00:00:00Z\",\"href\":\"https://api.onfido.com/v3/checks/check-123\"}}}";
+      "{\"payload\":{\"resource_type\":\"check\",\"action\":\"check.completed\",\"object\":{\"id\":\"check-123\",\"status\":\"complete\",\"completed_at_iso8601\":\"2020-01-01T00:00:00Z\",\"href\":\"https://api.onfido.com/v3.1/checks/check-123\"}}}";
```
Comment (ciaran16) [inline onfido-java/src/test/java/com/onfido/webhooks/WebhookEventVerifierTest.java:13]: This href and the one below won't be changing - should also fix your failing test too as the signature isn't valid if the contents are changed.

File: onfido-java/src/test/java/com/onfido/integration/CheckManagerTest.java [hunk 2]
```diff
@@ -97,4 +98,22 @@ public void resumeCheck() throws Exception {
...
   }
+
+  @Test
+  public void downloadCheck() throws Exception {
+    MockWebServer server = mockFileRequestResponse("test", "application/pdf");
+
+    Onfido onfido =
+        Onfido.builder().apiToken("token").unknownApiUrl(server.url("/").toString()).build();
+
+    FileDownload download = onfido.check.download("check id");
```
Comment (ciaran16) [inline onfido-java/src/test/java/com/onfido/integration/CheckManagerTest.java:109]: Minor: I'd use `check_id` for this so we can test the exact same id gets used in the path (without the %20)

**Comment by superdevx64** [comment]
@JakeBat we've had to make some changes here, do you mind doing another review since it's different enough now from the last one you did. Apologies.

**Review by JakeBat — APPROVED** [review]

File: onfido-java/src/main/java/com/onfido/Onfido.java [hunk 3]
```diff
@@ -63,7 +63,7 @@ private Onfido(Builder builder) {
     /** The Api token. */
     public String apiToken = "";
     /** The Api url. */
-    public String apiUrl = DEFAULT_API_URL;
+    public String apiUrl = EU_API_URL;
```
Comment (superdevx64) [inline onfido-java/src/main/java/com/onfido/Onfido.java:66]: @ciaran16 @JakeBat in hindsight, if we don't want a default, presumably this should be `"";`?

Comment (JakeBat) [reply onfido-java/src/main/java/com/onfido/Onfido.java:66]: Yeah, that would make sense, if we leave it this way, my new check would be pointless, unless Ciaran objects I would say update it.

**Review by ciaran16 — APPROVED** [review]

**Pull request merged**